### PR TITLE
Feature/listen for contracts created

### PR DESCRIPTION
**Pull request by FredrikAreschoug**

### CODE REVIEW

File: member-service/src/main/java/com/hedvig/memberservice/web/v2/dto/SignStatusResponse.java [hunk 1]
```diff
@@ -27,7 +27,12 @@ public static SignStatusResponse CreateFromEntity(
             ? null
             : new CollectData(collectResponse.getStatus(), collectResponse.getHintCode());
 
-    return new SignStatusResponse(session.getStatus(), data);
+    SignStatus status = session.getStatus();
+    if (status == SignStatus.COMPLETED) {
+      status = session.getIsContractsCreated() ? SignStatus.COMPLETED : SignStatus.IN_PROGRESS;
+    }
+
+    return new SignStatusResponse(status, data);
```
Comment (vonElfvin) [inline member-service/src/main/java/com/hedvig/memberservice/web/v2/dto/SignStatusResponse.java:35]:
```suggestion
    SignStatus status = session.getStatus();
    if (status == SignStatus.COMPLETED) {
      status = contractService. hasContract(memberId) ? SignStatus.COMPLETED : SignStatus.IN_PROGRESS;
    }

    return new SignStatusResponse(status, data);
```

File: member-service/src/main/java/com/hedvig/memberservice/services/SigningService.kt [hunk 2]
```diff
@@ -103,12 +112,33 @@ class SigningService(
         }
     }
 
+    @Transactional
+    fun scheduleContractsCreatedJob(memberId: Long, signMethod: SignMethod) {
+        try {
+            val jobName = "POLL_CONTRACTS_CREATE_FOR ${memberId}_JOB"
```
Comment (johantjellden) [inline member-service/src/main/java/com/hedvig/memberservice/services/SigningService.kt:118]: Missing '_'

File: member-service/src/main/java/com/hedvig/memberservice/web/InternalMembersController.java [hunk 3]
```diff
@@ -39,13 +40,16 @@
   private final MemberRepository memberRepository;
   private final MemberQueryService memberQueryService;
   private final TraceMemberService traceMemberService;
+  private final SigningService signingService;
```
Comment (johantjellden) [inline member-service/src/main/java/com/hedvig/memberservice/web/InternalMembersController.java:43]: Maybe we don't need this anymore?

File: member-service/src/main/java/com/hedvig/memberservice/web/v2/dto/SignStatusResponse.java [hunk 4]
```diff
@@ -27,7 +26,12 @@ public static SignStatusResponse CreateFromEntity(
             ? null
             : new CollectData(collectResponse.getStatus(), collectResponse.getHintCode());
 
-    return new SignStatusResponse(session.getStatus(), data);
+    SignStatus status = session.getStatus();
+    if (status == SignStatus.COMPLETED) {
+      status = session.getHasContract() ? SignStatus.COMPLETED : SignStatus.IN_PROGRESS;
+    }
```
Comment (johantjellden) [inline member-service/src/main/java/com/hedvig/memberservice/web/v2/dto/SignStatusResponse.java:32]: Maybe we should move this login into signsession. Something like `getSignAndContractStatus`?

File: member-service/src/test/java/com/hedvig/memberservice/InternalMembersControllerTests.java [hunk 5]
```diff
@@ -45,6 +46,8 @@
 
   @MockBean private TraceMemberService traceMemberService;
 
+  @MockBean private SigningService signingService;
```
Comment (johantjellden) [inline member-service/src/test/java/com/hedvig/memberservice/InternalMembersControllerTests.java:49]: If we remove this from the controlle we can remove it here.

**Pull request merged**